openmc/tallies.py

```diff
@@ -1468,32 +1528,20 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Query the mean and std dev so the tally data is read in from file
+        # if it has not already been read in.
+        self.mean, self.std_dev, other.mean, other.std_dev
```

Comment:
Sounds like a plan, just create an issue for this this so we don't forget to implement the change in the near future.